pravega/flink-connectors
# [issue-335] Change to the new DataType in the Table API

Signed-off-by: Brian Zhou <brian.zhou@emc.com>

**Note**
Please review after #297 is merged, this PR is based on that branch

**Change log description**
Since Flink 1.9, Flink introduced a new Type system `DataType` specifically for Table API to replace the legacy general `TypeInformation`. This is a fix to change to the the `DataType` in Table API.
It also contains many other small fixes to finalize the 1.10 implementation including
- Make `writer.enable-watermark` optional
- Generated column issue similar to [FLINK-15191](https://issues.apache.org/jira/browse/FLINK-15191)

**Purpose of the change**
Fixes #335 

**What the code does**
- Fix the implementation of `emitDataStream` method by reading Flink's implementation 
- Change all the `Types` and `TypeInformation` used in deprecated API, use the new interface instead. 
- Remove the "abstract" of the classes `FlinkPravegaTableSource` and `FlinkPravegaTableSink` as the actually implementations of the abstract class are of the same behaviour.
- Replace the `taskConfiguration` into `TableSchema` in the sink as a best practice described in TableSink interface.
- Use the physical schema to fix the generated column issue similar with [FLINK-15191](https://issues.apache.org/jira/browse/FLINK-15191)
- Update the test and fix the timestamp issue in IT case by bridging

**How to verify it**
`./gradlew clean build` passes
Verified under the sql demos in pravega-samples, see https://github.com/pravega/pravega-samples/pull/246 
Should cherry-pick to `r0.7`
